### ISSUE TITLE
I think there is an issue with computed properties not triggering on findAll data when findRecord is called

### ISSUE THREAD

**Issue by KroneckerDeIta** [emberjs/data#8411]
### Description

Computed properties do not seem to be triggered when findRecord is called (see example code below).

```cli
import Controller from '@ember/controller';
import { service } from '@ember/service';
import { later } from '@ember/runloop';
import { computed } from '@ember/object';
import { tracked } from "@glimmer/tracking";

export default class ApplicationController extends Controller {
  @service store;
  @tracked movies;

  @computed('movies.@each.name')
  get moviesComputed() {
    // Not being called when findRecord is called below.
    let moviesString = "";

    this.get("movies")?.forEach((movie) => {
      moviesString += movie.name + ","
    });

    return moviesString;
  }

  constructor() {
    super(...arguments);

    this.store.findAll('movie').then((movies) => {
      this.movies = movies;
    });

    later(() => {
      this.store.findRecord('movie', 4);
    }, 5000);
  }
}
```

What does work:

- If the computed property is converted to a getter.
- `{{#each}}` handlebars show the update.
- If I call `this.notifyPropertyChange("movies");` in a finally block of the findRecord.

Our problem:

This is an example I have come up with for native classes, but it is also affecting classic Ember objects as well. We sometimes store the passed result from the findAll in a service as it is used by more than one page and expect computed properties to pick up updates from findRecord requests (and other calls that would change the list). This worked for ember-cli 3.28.6 and ember-data 3.28.4, but doesn't seem to work for ember-cli 4.8.0 and ember-data 4.7.3.

### Versions

```cli
ember-data-records-issue
├─┬ @ember/test-helpers@2.9.3
│ ├─┬ @embroider/util@1.10.0
│ │ └── ember-source@4.8.4 deduped
│ └── ember-source@4.8.4 deduped
├─┬ ember-qunit@6.1.1
│ └── ember-source@4.8.4 deduped
└── ember-source@4.8.4

ember-data-records-issue
├─┬ ember-cli-dependency-checker@3.3.1
│ └── ember-cli@4.8.0 deduped
└── ember-cli@4.8.0

ember-data-records-issue
├─┬ ember-cli-mirage@2.4.0
│ └── ember-data@4.7.3 deduped
└── ember-data@4.7.3
```

**Comment by KroneckerDeIta** [comment]
I just tried to convert a service to be native class, but realize we are using a sort computed property and that is not picking up the findRecord change to the findAll list either (I guess obviously - it is a computed property).

**Comment by KroneckerDeIta** [comment]
For the moment I am having to create a copy (using slice) and store the list returned from the findAll promise response and call addObject on that list in the promise response from a findRecord and removeObject on a delete. I'm surprised no other issues have been raised relating to this issue though?

**Comment by runspired** [comment]
you should never use `computed` in modern ember anyway, these should be just getters and if memoization is needed use `cached` to decorate them

That said, `length` on these record arrays is still tracked in a manner that the computeds will still work. What's your ember-source version and whats the status of the `default-async-observers` feature flag?

**Comment by runspired** [comment]
also I'd try `4.8.x` vs `4.7.x` as a landing target, 4.8 gets more support as an lts and has more fixes

**Comment by runspired** [comment]
@richgt this may actually be the thing you hit in 3.28 too 🤔

**Comment by KroneckerDeIta** [comment]
Thanks for the response and looking into this and fixing. I know this is closed but can I ask what versions this fix will be present in? I have updated my projects to the latest LTS of ember-cli (4.8.0), which by default comes with ember-data (~4.7.3) when creating a new project. I did just try to update just ember-data to ~4.8.6 (just to see if the project would work with a newer version, not because I thought the fix might be in there) but I got an exception when running `ember s`:

`@embroider/macros/runtime.js: @babel/template placeholder "SAMPLE_FEATURE_FLAG": Expected string substitutionError: @babel/template placeholder "SAMPLE_FEATURE_FLAG": Expected string substitution`

Basically, will this fix make it's way into a version of ember data that the ember-cli 4.8.x series will support?

**Comment by runspired** [comment]
@KroneckerDeIta this fix is in 4.8.7, which is LTS. 4.7 is not supported.

**Comment by runspired** [comment]
As far as the build error goes, probably you are hitting this: https://github.com/embroider-build/embroider/issues/1077

which would indicate you need to update some dependencies that have older versions of embroider/macros as a dependency or force all embroider/macros usage onto a more recent version via an override (this is broadly speaking fairly safe to do)

**Comment by KroneckerDeIta** [comment]
Ah it was the one package (ember-cli-mirage) I had added to project I was testing with:

https://github.com/miragejs/ember-cli-mirage/issues/2461

Will set embroider/macros to a later version. Thanks
